internal/controllers/certificates/cert_manager.go

```diff
@@ -0,0 +1,188 @@
+// Copyright 2021 tsuru authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package certificates
+
+import (
+	"context"
+	"fmt"
+
+	cmv1 "github.com/jetstack/cert-manager/pkg/apis/certmanager/v1"
+	cmmeta "github.com/jetstack/cert-manager/pkg/apis/meta/v1"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/tsuru/rpaas-operator/api/v1alpha1"
+)
+
+const CertManagerCertificateName string = "cert-manager"
+
+func reconcileCertManager(ctx context.Context, client client.Client, instance *v1alpha1.RpaasInstance) error {
+	if instance.Spec.DynamicCertificates == nil || instance.Spec.DynamicCertificates.CertManager == nil {
+		return deleteCertManager(ctx, client, instance)
+	}
+
+	issuer, err := getCertManagerIssuer(ctx, client, instance)
+	if err != nil {
+		return err
+	}
+
+	cert, err := getCertificate(ctx, client, instance)
+	if err != nil && k8serrors.IsNotFound(err) {
+		return client.Create(ctx, newCertificate(instance, issuer))
+	}
+
+	newCert := newCertificate(instance, issuer)
+	newCert.ResourceVersion = cert.ResourceVersion
+
+	if err = client.Update(ctx, newCert); err != nil {
+		return err
+	}
+
+	return reconcileCertificateSecret(ctx, client, instance, cert)
+}
+
+func deleteCertManager(ctx context.Context, client client.Client, instance *v1alpha1.RpaasInstance) error {
+	cert, err := getCertificate(ctx, client, instance)
+	if err != nil && k8serrors.IsNotFound(err) {
+		return nil
+	}
+
+	if err != nil {
+		return err
+	}
+
+	certmanagerSecretName := cert.Spec.SecretName
+
+	if err = client.Delete(ctx, cert); err != nil {
+		return err
+	}
+
+	var s corev1.Secret
+	err = client.Get(ctx, types.NamespacedName{
+		Name:      certmanagerSecretName,
```

Comment:
why don't you just use cert.Spec.SecretName? I see no gain from the attribution in line 60.